plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestAvroConfluentContentSchemaReader.java

```diff
@@ -51,6 +56,32 @@ public void testAvroConfluentSchemaReader()
                 .hasMessage("Could not resolve schema for the 'another-schema' subject");
     }
 
+    @Test
+    public void getAvroSchemaWithReferences()
+            throws Exception
+    {
+        MockSchemaRegistryClient mockSchemaRegistryClient = new MockSchemaRegistryClient();
+        int schemaId = mockSchemaRegistryClient.register("base_schema-value", new AvroSchema(getAvroSchema()));
+        ParsedSchema schemaWithReference = mockSchemaRegistryClient.parseSchema(null, getAvroSchemaWithReference(), ImmutableList.of(new SchemaReference(TOPIC, "base_schema-value", schemaId)))
+                .orElseThrow();
+        mockSchemaRegistryClient.register(SUBJECT_NAME, schemaWithReference);
+
+        AvroConfluentContentSchemaReader avroConfluentSchemaReader = new AvroConfluentContentSchemaReader(mockSchemaRegistryClient);
+        assertThat(avroConfluentSchemaReader.readSchema(Optional.empty(), Optional.of(SUBJECT_NAME)).map(schema -> new Parser().parse(schema))).isPresent();
+    }
+
+    private static String getAvroSchemaWithReference()
+    {
+        return "{\n" +
+                "    \"type\":\"record\",\n" +
+                "    \"name\":\"Schema2\",\n" +
+                "    \"fields\":[\n" +
```

Comment:
How does it work? You referenced a schema and then you specified fields, does it mean that actual fields will be an union of these two? Or is it only about that type `test` is now defined and you can reuse it?

Where is type `test` defined?

Reply:
In some cases it can be union. But here it would be nested. `test` type is referred from the `SchemaRegistry`.